presto-spi/src/main/java/com/facebook/presto/spi/page/PagesSerdeUtil.java

```diff
@@ -60,6 +67,17 @@ public static void writeSerializedPage(SliceOutput output, SerializedPage page)
         output.writeByte(page.getPageCodecMarkers());
         output.writeInt(page.getUncompressedSizeInBytes());
         output.writeInt(page.getSizeInBytes());
+        if (CHECKSUMMED.isSet(page.getPageCodecMarkers())) {
+            output.writeLong(page.getChecksum());
+            try {
+                String host = Inet6Address.getLocalHost().getHostAddress();
```

Comment:
`Inet6Address.getLocalHost().getHostAddress()` is actually quite expensive. Could potentially be too expensive to be done for every single page.

Also are we sure this is the right way of extracting a hostname? Wouldn't it return a `127.0.0.1` analogue in IPv6?

CC: @bhhari @tdcmeehan

Reply:
This is the right way to get hostname, I have tested it, it will give twshared...